### ISSUE TITLE
Bug: [no-import-type-side-effects] doesn't allow side effects only imports

### ISSUE THREAD

**Issue by chbdetta** [typescript-eslint/typescript-eslint#6431]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=JYWwDg9gTgLgBAchgUwM4wHQGNWoUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1soFtiPodIqVrJKAE0T0AZtMRl8qDACZwYAL4h1QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
import 'test.css'
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-import-type-side-effects": 2,
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

no error

### Actual Result

error, and auto fixed to wrong code:
```
import type 'test.css'
```

**Comment by bradzacher** [comment]
Ironic that it warns you about explicit side-effect imports due this bug!

Also I can't believe that I didn't include a `node.specifiers.length > 0` check 🤦

**Comment by JoshuaKGoldberg** [comment]
D'oh, I forgot about those...
